tests/Agent/IntegrationTests/UnboundedIntegrationTests/MySqlConnectorTests.cs

```diff
@@ -0,0 +1,201 @@
+// Copyright 2020 New Relic, Inc. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using NewRelic.Agent.IntegrationTestHelpers;
+using NewRelic.Agent.IntegrationTestHelpers.Models;
+using NewRelic.Agent.IntegrationTests.Shared;
+using NewRelic.Testing.Assertions;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace NewRelic.Agent.UnboundedIntegrationTests
+{
+    [CollectionDefinition("MySqlConnector", DisableParallelization = true)]
```

Comment:
Did you run into some problems when these tests where running in parallel? For example, with our .Net core test apps, we can't run multiple `dotnet publish` commands on a single test app concurrently because it results in publishing errors. Was there some shared resource in the database that caused a problem? I'm mostly asking because we have the other MySql tests and are wondering if they will also interfere with these tests?

Reply:
I was running the tests through the R# Unit Test Runner in VS2019. I would get an error from the fixture setup (sorry for light on details, this is from memory) from `StartWebServer`: https://github.com/newrelic/newrelic-dotnet-agent/blob/2b8e8cdc71146b1196e6a455e981be37fe789532/tests/Agent/IntegrationTests/HostedWebCore/HostedWebCore.cs#L70

The HRESULT converted to an Exception was something like "Cannot access the file when it's already in use". I tried using ProcMon to see what file might have a sharing violation but it didn't reveal anything. So I started to wonder if multiple instances of `ClassFixture<RemoteServiceFixtures.MySqlConnectorBasicMvcFixture>` were being instantiated at the same time (one per test class) and stomping on each other.

It seemed like you had multiple test classes that all used the same fixture presumably successfully (e.g., all the Postgres tests) so it may be the R# test runner that's causing the problem? I wanted to try using some different test runners and raising the issue directly if I was still getting blocked on it.

For now, this was a workaround I was using to run and debug the tests, but I'd like to remove it if it's not truly necessary.

Reply:
Here's the test failure I got:

```
Test Name:	NewRelic.Agent.UnboundedIntegrationTests.MySqlConnectorExecuteNonQueryTest.Test
Test Outcome:	Failed
Result Message:	
Remote process never generated a .pid file!
Expected: True
Actual:   False
Result StandardOutput:	
IntegrationTestAppName
Test HomeC:\IntegrationTestWorkingDirectory\BasicMvcApplication_a921ef30-eca8-4175-86a7-cce2037547bd\newrelichome

====== [RemoteWebApplication]: WaitForHostedWebCoreToStartListening standard output log =====
[9/16/2020 8:19:58 PM 45524-1] HostedWebCore: Firing up...args: --port=54687
[9/16/2020 8:19:58 PM 45524-1] HostedWebCore: Starting directory: C:\IntegrationTestWorkingDirectory\BasicMvcApplication_a921ef30-eca8-4175-86a7-cce2037547bd\HostedWebCore
[9/16/2020 8:19:58 PM 45524-1] HostedWebCore: Environment Variables:   [snip]
[9/16/2020 8:19:58 PM 45524-1] HostedWebCore: Starting server...
[9/16/2020 8:19:59 PM 45524-1] HostedWebCore: HostedWebCore.exe failed: Reason unknown.: System.IO.FileLoadException: The process cannot access the file because it is being used by another process. (Exception from HRESULT: 0x80070020)
   at System.Runtime.InteropServices.Marshal.ThrowExceptionForHRInternal(Int32 errorCode, IntPtr errorInfo)
   at System.Runtime.InteropServices.Marshal.ThrowExceptionForHR(Int32 errorCode)
   at HostedWebCore.HostedWebCore.StartWebServer() in C:\newrelic-dotnet-agent\tests\Agent\IntegrationTests\HostedWebCore\HostedWebCore.cs:line 71
   at HostedWebCore.HostedWebCore.Run() in C:\newrelic-dotnet-agent\tests\Agent\IntegrationTests\HostedWebCore\HostedWebCore.cs:line 60
   at HostedWebCore.Program.Main(String[] args) in C:\newrelic-dotnet-agent\tests\Agent\IntegrationTests\HostedWebCore\Program.cs:line 50
```

Reply:
VS Test Explorer can run these tests in parallel just fine (I see it spinning up six separate copies of HostedWebCore.exe) so I'll delete these attributes and just use Test Explorer instead. (The ReSharper Unit Test Sessions runner seems to have problems running more than one test simultaneously, it seems?)

Reply:
I have definitely seen that error before, and I have yet to figure out what the cause is. It only seems to happen occasionally for me. I've seen it happen locally when using Test Explorer, and I've seen it when running the tests using our Integration Tests script. I also thought it had something to do with trying to start up multiple instances of the same app at the same time. There might be something we can do in the HostWebCore app to protect some api call(s) that we make so that we only call it for one application at a time, but we haven't figured out how to reliably reproduce that problem yet.